src/components/Modals/QrCode/Form.tsx

```diff
@@ -0,0 +1,152 @@
+import type { AccountId, AssetId, ChainId } from '@shapeshiftoss/caip'
+import { ethAssetId } from '@shapeshiftoss/caip'
+import type { FeeDataEstimate } from '@shapeshiftoss/chain-adapters'
+import { FeeDataKey } from '@shapeshiftoss/chain-adapters'
+import { AnimatePresence } from 'framer-motion'
+import { useCallback, useEffect } from 'react'
+import { FormProvider, useForm } from 'react-hook-form'
+import { Redirect, Route, Switch, useHistory, useLocation } from 'react-router-dom'
+import { QrCodeScanner } from 'components/QrCodeScanner/QrCodeScanner'
+import { SelectAssetRouter } from 'components/SelectAssets/SelectAssetRouter'
+import { parseMaybeUrl } from 'lib/address/address'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import {
+  selectAssetById,
+  selectMarketDataById,
+  selectSelectedCurrency,
+} from 'state/slices/selectors'
+import { store, useAppSelector } from 'state/store'
+
+import { useFormSend } from '../Send/hooks/useFormSend/useFormSend'
+import { SendFormFields, SendRoutes } from '../Send/SendCommon'
+import { Address } from '../Send/views/Address'
+import { Confirm } from '../Send/views/Confirm'
+import { Details } from '../Send/views/Details'
+
+export type SendInput<T extends ChainId = ChainId> = {
+  [SendFormFields.AccountId]: AccountId
+  [SendFormFields.To]: string
+  [SendFormFields.From]: string
+  [SendFormFields.AmountFieldError]: string | [string, { asset: string }]
+  [SendFormFields.AssetId]: AssetId
+  [SendFormFields.CryptoAmount]: string
+  [SendFormFields.EstimatedFees]: FeeDataEstimate<T>
+  [SendFormFields.FeeType]: FeeDataKey
+  [SendFormFields.FiatAmount]: string
+  [SendFormFields.FiatSymbol]: string
+  [SendFormFields.Input]: string
+  [SendFormFields.Memo]?: string
+  [SendFormFields.SendMax]: boolean
+  [SendFormFields.VanityAddress]: string
+}
+
+type QrCodeFormProps = {
+  assetId?: AssetId
+  accountId?: AccountId
+}
+
+export const Form: React.FC<QrCodeFormProps> = ({ accountId }) => {
+  const location = useLocation()
+  const history = useHistory()
+  const { handleFormSend } = useFormSend()
+  const selectedCurrency = useAppSelector(selectSelectedCurrency)
+
+  const methods = useForm<SendInput>({
+    mode: 'onChange',
+    defaultValues: {
+      accountId,
+      to: '',
+      vanityAddress: '',
+      assetId: '',
+      feeType: FeeDataKey.Average,
+      cryptoAmount: '',
+      fiatAmount: '',
+      fiatSymbol: selectedCurrency,
+    },
+  })
+
+  const handleAssetSelect = useCallback(
+    (assetId: AssetId) => {
+      const asset = selectAssetById(store.getState(), assetId ?? '')
```

Comment:
maybe select assets outside callback and inject those in rather than `getState()` in here?